### ISSUE TITLE
[iOS] Error to build app with expo-linear-gradient 11.3.0

### ISSUE THREAD

**Issue by TomCorvus** [expo/expo#17252]
### Summary

Hi 👋

Since I updated `expo-linear-gradient` to `11.3.0` version, I have an error when I try to build the app in debug mode with Xcode.
```
/Users/nameuser/Documents/projects/mobile/packages/appname/node_modules/expo-linear-gradient/ios/LinearGradientModule.swift:12:9: Cannot convert value of type 'LinearGradientView' to closure result type 'UIView'
/Users/nameuser/Documents/projects/mobile/packages/appname/node_modules/expo-linear-gradient/ios/LinearGradientModule.swift:15:7: Global function 'prop' requires that 'LinearGradientView' inherit from 'UIView'
/Users/nameuser/Documents/projects/mobile/packages/appname/node_modules/expo-linear-gradient/ios/LinearGradientModule.swift:19:7: Global function 'prop' requires that 'LinearGradientView' inherit from 'UIView'
/Users/nameuser/Documents/projects/mobile/packages/appname/node_modules/expo-linear-gradient/ios/LinearGradientModule.swift:23:7: Global function 'prop' requires that 'LinearGradientView' inherit from 'UIView'
/Users/nameuser/Documents/projects/mobile/packages/appname/node_modules/expo-linear-gradient/ios/LinearGradientModule.swift:27:7: Global function 'prop' requires that 'LinearGradientView' inherit from 'UIView'
/Users/nameuser/Documents/projects/mobile/packages/appname/node_modules/expo-linear-gradient/ios/LinearGradientView.swift:7:22: Property does not override any property from its superclass
/Users/nameuser/Documents/projects/mobile/packages/appname/node_modules/expo-linear-gradient/ios/LinearGradientView.swift:6:33: Cannot find type 'ExpoView' in scope
/Users/nameuser/Documents/projects/mobile/packages/appname/node_modules/expo-linear-gradient/ios/LinearGradientView.swift:12:12: Cannot find 'layer' in scope
```

Did I miss something?

Thanks

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

iOS

### Environment

```

  expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 12.3.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.17.6 - /usr/local/bin/node
      Yarn: 3.2.0 - /usr/local/bin/yarn
      npm: 8.6.0 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8193401
      Xcode: 13.3.1/13E500a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.6 => 44.0.6 
      react: ^17.0.2 => 17.0.2 
      react-native: ~0.68.1 => 0.68.1 
    npmGlobalPackages:
      eas-cli: 0.8.1
      expo-cli: 5.4.3
    Expo Workflow: bare
```

### Reproducible demo

Just run:
`expo init my-app` with Bare workflow minimal template.
Install `expo-linear-gradient@11.3.0`
run `expo run:ios`

You will get 8 errors normally.

**Comment by barthap** [comment]
Did you update the `expo-modules-core` package too? Which version do you have?

**Comment by TomCorvus** [comment]
I kept the defined expo-modules-core version in Expo 44.0.6 dependencies.
`expo-modules-core: 0.6.5`

**Comment by TomCorvus** [comment]
Thanks @barthap, the problem is fixed by setting `0.9.0` version of `expo-modules-core`

**Comment by Zernach** [comment]
Or, in my case, the problem was solved by downgrading `expo-linear-gradient` to `11.0.3` 😄 per @barthap's first option... because the app I'm currently developing is still running on expo 44. Thank you!
